esp-wifi/src/esp_now/mod.rs

```diff
@@ -480,42 +478,182 @@ impl<'d> EspNow<'d> {
+    pub fn get_version(&self) -> Result<u32, EspNowError> {
+        self.manager.get_version()
+    }
+
+    /// Add a peer to the list of known peers
+    pub fn add_peer(&self, peer: PeerInfo) -> Result<(), EspNowError> {
+        self.manager.add_peer(peer)
+    }
+
+    /// Remove the given peer
+    pub fn remove_peer(&self, peer_address: &[u8; 6]) -> Result<(), EspNowError> {
+        self.manager.remove_peer(peer_address)
+    }
+
+    /// Modify a peer information
+    pub fn modify_peer(&self, peer: PeerInfo) -> Result<(), EspNowError> {
+        self.manager.modify_peer(peer)
+    }
+
+    /// Get peer by MAC address
+    pub fn get_peer(&self, peer_address: &[u8; 6]) -> Result<PeerInfo, EspNowError> {
+        self.manager.get_peer(peer_address)
+    }
+
+    /// Fetch a peer from peer list
     ///
-    /// Note: if you firstly dropped waiter of a sending and then wait for a following sending,
-    /// you probably get unreliable status because we cannot determine which sending the waited status
-    /// belongs to.
-    pub fn wait(self) -> SendStatus {
-        while !ESP_NOW_SEND_CB_INVOKED.load(Ordering::Acquire) {}
+    /// Only returns peers which address is unicast, for multicast/broadcast addresses, the function will
+    /// skip the entry and find the next in the peer list.
+    pub fn fetch_peer(&self, from_head: bool) -> Result<PeerInfo, EspNowError> {
+        self.manager.fetch_peer(from_head)
+    }
 
-        if ESP_NOW_SEND_STATUS.load(Ordering::Relaxed) {
-            SendStatus::Success
-        } else {
-            SendStatus::Failed
+    /// Check is peer is known
+    pub fn peer_exists(&self, peer_address: &[u8; 6]) -> bool {
+        self.manager.peer_exists(peer_address)
+    }
+
+    /// Get the number of peers
+    pub fn peer_count(&self) -> Result<PeerCount, EspNowError> {
+        self.manager.peer_count()
+    }
+
+    /// Set the primary master key
+    pub fn set_pmk(&self, pmk: &[u8; 16]) -> Result<(), EspNowError> {
+        self.manager.set_pmk(pmk)
+    }
+
+    /// Set wake window for esp_now to wake up in interval unit
+    ///
+    /// Window is milliseconds the chip keep waked each interval, from 0 to 65535.
+    pub fn set_wake_window(&self, wake_window: u16) -> Result<(), EspNowError> {
+        self.manager.set_wake_window(wake_window)
+    }
+
+    /// Config ESPNOW rate
+    pub fn set_rate(&self, rate: WifiPhyRate) -> Result<(), EspNowError> {
+        self.manager.set_rate(rate)
+    }
+
+    /// Send data to peer
+    ///
+    /// The peer needs to be added to the peer list first.
+    pub fn send<'s>(&'s mut self, dst_addr: &[u8; 6], data: &[u8]) -> Result<SendWaiter<'s>, EspNowError> {
+        self.sender.send(dst_addr, data)
+    }
+
+    /// Receive data
+    pub fn receive(&self) -> Option<ReceivedData> {
+        self.receiver.receive()
+    }
+}
+
+impl Drop for EspNowSender<'_> {
+    fn drop(&mut self) {
+        if ESP_NOW_RC.fetch_sub(1, Ordering::Release) == 1 {
```

Comment:
I think it would be a bit nicer if these structs contained a type (EspNowRc for example, maybe?) that is responsible for this cleanup. Then you would only need a single Drop implementation for that type instead of three identical ones.

Reply:
Good idea. I added an EspNowRc in [98a4f0e](https://github.com/esp-rs/esp-wifi/pull/240/commits/98a4f0e9dd115aef0745d0dbc648a999c4009cbf)